minima/database.nim

```diff
@@ -20,38 +20,55 @@ type
         KeyNotFound             = "minima: key not found"
         PersistenceFailed       = "minima: persistence failed"
 
-proc log(db: Database, key: seq[byte], value: seq[byte]) =
-    let write = concat(
-        @(uint32(len(key)).toBytes),
-        @(uint32(len(value)).toBytes),
-        key,
-        value
-    )
-
-    # @TODO CATCH EXCEPTIONS
-    discard db.log.writeBytes(write, 0, len(write))
-    db.log.flushFile()
+proc toAESKey*(str: string): array[aes256.sizeKey, byte] =
+    ## Concerts a string to an AES Key for opening an encrypted database.
+    ## 
+    ## **Example::**
+    ## .. code-block::
+    ##   let password = "foo".toAESKey
+    var pass = str
+    var key: array[32, byte]
+    copyMem(addr key[0], addr pass[0], len(pass))
 
-proc readInt(file: File): int =
-    var arr: array[4, byte]
-    discard file.readBytes(arr, 0, 4)
+    return key
 
-    return int(uint32.fromBytes(arr))
+proc open*(dir: string, key: array[aes256.sizeKey, byte]): Result[Database, DatabaseError] =
+    ## Opens an encrypted database at the specified path.
+    ## 
+    ## **Example:**
+    ##
+    ## .. code-block::
+    ##   let result = open("/tmp/minima", "password".toAESKey)
+    ##   if not result.isOk:
+    ##     return
+    if not os.existsDir(dir):
+        try:
+            os.createDir(dir)
+        except:
+            return err(DirectoryCreationFailed)
 
-proc recover(db: Database) =
-    while db.log.getFilePos() <= db.log.getFileSize() - 1:
-        var keyLen = readInt(db.log)
-        var valLen = readInt(db.log)
+    var path = dir & "/minima.db"
```

Comment:
Use the `/` operator when combining paths. This will produce better cross-platform code (some OSes use different path separators).

```nim
var path = dir / "minima.db"
```